From 9d0640874425e9f3f265c9baff7a47139b25ea7d Mon Sep 17 00:00:00 2001
From: Keith Packard <keithp@keithp.com>
Date: Thu, 14 Jan 2021 17:54:22 -0800
Subject: [PATCH 1/2] libc: Remove #include <sys/select.h> from sys/types.h

picolibc's sys/select.h is likely to be replaced by the underlying
operating system version (as it is on Zephyr). Don't include it from
sys/types.h as that version may depend on other definitions in
sys/types.h which haven't yet been defined.

Signed-off-by: Keith Packard <keithp@keithp.com>
---
 newlib/libc/include/sys/types.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/newlib/libc/include/sys/types.h b/newlib/libc/include/sys/types.h
index ea25222c2..1a0abcb83 100644
--- a/newlib/libc/include/sys/types.h
+++ b/newlib/libc/include/sys/types.h
@@ -75,7 +75,6 @@ typedef __intptr_t register_t;
 
 #if __BSD_VISIBLE
 #include <machine/endian.h>
-#include <sys/select.h>
 #  define	physadr		physadr_t
 #  define	quad		quad_t
 
-- 
2.30.0

